templates/dockerize.yml

```diff
@@ -0,0 +1,74 @@
+# This template is expects exactly ONE Dockerfile in the microservice repo
+# This template utilizes an external dependency called mvn-version to extract
+# the version of the Maven pom.xml file, which it expects to be in the main
+# directory for the microservice, with a foldername that is the same as the
+# image name that will be used for the docker image
+
+parameters:
+
+  # project configuration parameters:
+- name: imageName                   # <-- e.g., project-registry-gateway
+  type: string
+- name: containerRepo               # <-- e.g., myproject.azurecr.io, myDockerHub
+  type: string
+- name: containerServiceConnection  # <-- name of the service connection associated
+  type: string                      #     with the above repo
+- name: containerUrl                # <-- URL to the associated repo
+  type: string
+- name: mvnVersionDir               # <-- the path to mvn-version
+  type: string
+- name: version                     # <-- the version is dynamically set using mvn-version
+  type: string                      #     so there is no need to pass this parameter into
+  default: undefined                #     the template; the default value could be any string
+
+  # external access parameters:
+- name: discordChannel
+  type: string
+- name: discordKey
+  type: string
+
+steps:
+- script: |
+    sudo apt-get install zsh -y
+    sudo apt-get install xmlstarlet -y
+    sudo bash autogen.sh
+    sudo ./configure
+    sudo make
+    sudo make install
+  workingDirectory: ${{ parameters.mvnVersionDir }}
+  displayName: Install mvn-version --> Task
```

Comment:
it'd make more sense to download the [already-built mvn-version](https://github.com/emcrisostomo/mvn-version#getting-mvn-version) rather than building it from source. and it arguably makes sense to have the agent do that itself, rather than have the file be part of our code repo.

also, you'll want to run apt-get update before the apt-get installs, and any time you have a multiline script where you want every command to succeed, you can either break it up into more steps, use more `&&`, or use `set -e`.

also, this step lacks some context; i'd link to mvn-version here so that it becomes clearer.

Reply:
The thinking in keeping the code in the repo was that any changes that MIGHT be made to mvn-version would not cause any problems with the pipeline. This may not be a huge concern working within our current time frame, but the decision was made with long-term stability in mind. 

We can back this down to just pull it in during the build if you think that's a better way to handle this.

Reply:
good thinking, but still, if we link to the specific release number on github, then that won't change; a new version would be at a different link.

Reply:
I have updated the template to pull mvn-version-1.2.1 from its gitHub instead of holding it in the repo.
I also edited the template to allow the user to override the dynamic versioning and specify a custom version.  We do not expect to use a 'latest' tag, but if we change our minds in the future, this would allow that to be automated.